packages/augur-core/src/contracts/trading/ERC20Proxy.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity 0.5.15;
+
+import 'ROOT/libraries/CloneFactory.sol';
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/token/IERC1155.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract ERC20ProxyNexus {
+    using SafeMathint256 for uint256;
+
+    address public target1155;
+    address proxyToClone;
+
+    // tokenId -> owner -> spender -> allowance
+    // The spender is whoever called the proxy contract, which is msg.sender in the proxy and _caller here in the nexus.
+    mapping (uint256 => mapping (address => mapping (address => uint256))) public allowances;
+    // tokenId -> proxy contract
+    mapping (uint256 => address) public proxies;
+
+    constructor(address _proxyToClone, address _target1155) public {
+        proxyToClone = _proxyToClone;
+        target1155 = _target1155;
+        // This contract will be able to do anything with the user's 1155 tokens.
+        IERC1155(target1155).setApprovalForAll(address(this), true);
+    }
+
+    function newERC20(uint256 _tokenId) public returns (ERC20Proxy) {
```

Comment:
It's probably fine to have this contract do the ERC20 generation, though the pattern we have elsewhere is to have a specific factory contract which is called by contracts like this to generate them. Just easier to know where things happen I guess

Reply:
Seems to work so I'm going to leave this as-is.